File: ZkTony-Api/db/src/system/models/customer.rs
use serde::Deserialize;

#[derive(Deserialize, Debug, Clone)]
pub struct CustomerSaveReq {
    pub id: String,
    pub name: String,
    pub phone: String,
    pub address: String,
    pub source: String,
    pub industry: String,
    pub remarks: Option<String>,
    pub create_by: Option<String>,
    pub create_time: Option<String>,
}

#[derive(Deserialize, Debug, Clone)]
pub struct CustomerDeleteReq {
    pub id: String,
}

#[derive(Deserialize, Debug, Clone)]
pub struct CustomerGetReq {
    pub id: Option<String>,
    pub name: Option<String>,
    pub phone: Option<String>,
    pub address: Option<String>,
}

File: ZkTony-Manager/app/src/main/kotlin/com/zktony/manager/data/remote/model/Customer.kt
package com.zktony.manager.data.remote.model

import com.zktony.www.common.extension.currentTime
import java.util.*

/**
 * @author: 刘贺贺
 * @date: 2023-02-16 16:33
 */
data class Customer(
    val id: String = UUID.randomUUID().toString(),
    val name: String = "",
    val phone: String = "",
    val address: String = "",
    val source: String = "",
    val industry: String = "",
    val remarks: String = "",
    val create_by: String = "",
    val create_time: String = currentTime()
)
